index.js

```diff
@@ -298,68 +370,105 @@ function cmpType (a, b) {
   return typeOrder[a.type] - typeOrder[b.type]
 }
 
-DB.prototype._onpt = function (pt, seen, cb) {
+// Given a node by its version, this collects the node itself, and also
+// recursively climbs all ways and relations that the node (or its referers)
+// are referred to by.
+// OsmVersion, { OsmVersion: Boolean } -> [OsmDocument]
+DB.prototype._collectNodeAndReferers = function (version, seenAccum, cb) {
   cb = once(cb || noop)
   var self = this
-  var link = pt.value.toString('hex')
-  if (has(seen, link)) return cb(null, [])
-  seen[link] = true
-  var res = [], added = {}, pending = 2
-  self.log.get(link, function (err, doc) {
-    if (doc && doc.value && doc.value.k && doc.value.v) {
-      addDoc(doc.value.k, link, doc.value.v)
-    }
-    if (--pending === 0) cb(null, res)
+  if (has(seenAccum, version)) return cb(null, [])
+  var res = [], added = {}, pending = 1
```

Comment:
I think we need more robust error handling in this function. Any error (e.g. notFound) will cause this to stop short, and the error will be swallowed because the query function doesn't do anything with the error.

I _think_ what we want to do is continue even if there is an error. But the way this is currently handled seems a little fragile.

One approach I've seen substack use in other modules is to define `errors = []` in the outer context here, then also in the outer context:

```
function done (err, res) {
  if (err) errors.push(err)
  if (--pending <= 0) cb(errors.length ? errors : null, res)
}
```

and replace al the `if (err)` and `if (--pending === 0)` with `done(err|null, res)` and make sure we call `done(res)` on line 437 at the end of `onLinks`.

PS. I think `--pending <= 0` is safest here, given the complexity of this code and somehow getting below zero, and given the problems some code like this recently caused for CloudFlare. https://blog.cloudflare.com/incident-report-on-memory-leak-caused-by-cloudflare-parser-bug/